app/src/main/java/com/nitramite/utils/BackupUtils.java

```diff
@@ -0,0 +1,182 @@
+package com.nitramite.utils;
+
+import android.annotation.SuppressLint;
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import com.nitramite.paketinseuranta.R;
+
+import org.jetbrains.annotations.NonNls;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Objects;
+
+import static com.nitramite.paketinseuranta.DatabaseHelper.DATABASE_NAME;
+
+public class BackupUtils {
+
+    @NonNls
+    private static final String TAG = "BackupUtils";
+
+    // Backup database
+    public static Backup backupDatabase(Context context) {
+        Backup backup = new Backup();
+        try {
+            FileInputStream dbFileInputStream = getDBFileInputStream(context, backup);
+            OutputStream dbFileOutputStream = getDBOutputStream(context, backup);
+
+            // Write new file out
+            byte[] buffer = new byte[1024];
+            int length;
+            while ((length = dbFileInputStream.read(buffer)) > 0) {
```

Comment:
Interesting syntax, I don't think I've never seen an assignment inside a while loop on Java. :open_mouth:

Reply:
Old code.. originating to Stackoverflow where you see everything you cannot imagine.